prediction_market_agent/agents/autogen_general_agent/social_agent.py

```diff
@@ -0,0 +1,154 @@
+        code_execution_config={
+            "last_n_messages": 1,
+            "work_dir": "tasks",
+            "use_docker": False,
+        },
+    )
+
+    writer = autogen.AssistantAgent(
+        name="Writer",
+        llm_config=llm_config,
+        system_message="""
+        You are a professional influencer, known for your insightful and engaging tweets.
+        You transform complex concepts into compelling narratives.
+        You should improve the quality of the content based on the feedback from the user.
+        You must always return only the tweet.
+        """,
+    )
+
+    critic = autogen.AssistantAgent(
+        name="Critic",
+        llm_config=llm_config,
+        system_message="""
+            You are a critic, known for your thoroughness and commitment to standards.
+            Your task is to scrutinize content for any harmful elements or regulatory violations, ensuring
+            all materials align with required guidelines.
+            References to betting and gambling are allowed.
+            """,
+    )
+
+    return {
+        AutogenAgentType.CRITIC: critic,
+        AutogenAgentType.WRITER: writer,
+        AutogenAgentType.USER: user_proxy,
+    }
+
+
+def build_social_media_text(model: str, bets: list[Bet]) -> str | None:
+    """
+    Builds a tweet based on the five markets on Omen that are closing soonest.
+
+    This function utilizes the writer and critic agents to generate the tweet content. It first initializes the
+    necessary agents based on the provided model. Then, it registers a chat between the writer and critic agents.
+
+    The tweet content is generated using a template that includes questions about each market's title and likelihood.
+    """
+
+    agents = build_agents(model)
+    user_proxy, writer, critic = (
+        agents[AutogenAgentType.USER],
+        agents[AutogenAgentType.WRITER],
+        agents[AutogenAgentType.CRITIC],
+    )
+    user_proxy.register_nested_chats(
+        [
+            {
+                "recipient": critic,
+                "message": reflection_message,
+                "summary_method": "last_msg",
+                "max_turns": 1,
+            }
+        ],
+        trigger=writer,
+    )
+
+    # ToDO - fetch agent reasoning from DB and construct better tweets
+    #  See https://github.com/gnosis/prediction-market-agent/issues/150
+
+    task = Template(INFLUENCER_PROMPT).substitute(
+        BETS=[BetInputPrompt.from_bet(bet) for bet in bets]
+    )
+
+    # in case we trigger repeated runs, Cache makes it faster.
+    with Cache.disk(cache_seed=42) as cache:
+        # max_turns = the maximum number of turns for the chat between the two agents. One turn means one conversation round trip.
+        res = user_proxy.initiate_chat(
+            recipient=writer,
+            message=task,
+            max_turns=2,
+            summary_method="last_msg",
+            cache=cache,
+        )
+
+    tweet = res.summary
+    return str(
```

Comment:
The signature of this method has return type `str | None`, but it looks like it always returns `str`. Am I missing something?

Reply:
It can also return None if no bets are found in the last 24h